diffrax/_brownian/tree.py

```diff
@@ -63,62 +64,113 @@
 _BrownianReturn = TypeVar("_BrownianReturn", bound=AbstractBrownianIncrement)
 
 
+# An internal dataclass that holds the rescaled Levy areas
+# in addition to the non-rescaled ones, for the purposes of
+# taking the difference between two Levy areas.
+class _AbstractLevyVal(eqx.Module):
+    dt: eqx.AbstractVar[Inexact[Array, ""]]
+    W: eqx.AbstractVar[Array]
+
+
+class _BMLevyVal(_AbstractLevyVal):
+    dt: Inexact[Array, ""]
+    W: Array
+
+
+class _AbstractSpaceTimeLevyVal(_AbstractLevyVal):
+    H: eqx.AbstractVar[Array]
+    bar_H: eqx.AbstractVar[Array]
+
+
+class _SpaceTimeLevyVal(_AbstractSpaceTimeLevyVal):
+    dt: Inexact[Array, ""]
+    W: Array
+    H: Array
+    bar_H: Array
+
+
+class _SpaceTimeTimeLevyVal(_AbstractSpaceTimeLevyVal):
+    dt: Inexact[Array, ""]
+    W: Array
+    H: Array
+    bar_H: Array
+    K: Array
+    bar_K: Array
+
+
 class _State(eqx.Module):
     level: IntScalarLike  # level of the tree
-    s: RealScalarLike  # starting time of the interval
+    s: Inexact[Array, ""]  # starting time of the interval
```

Comment:
Times should always be real, not complex.

Reply:
I was trying to avoid a myriad of `with jax.numpy_dtype_promotion("standard"):`, which spaghettify the code a bit, and I thought the times which are only used internally could safely be complex. But I guess it could be a source of bugs, so fair enough.